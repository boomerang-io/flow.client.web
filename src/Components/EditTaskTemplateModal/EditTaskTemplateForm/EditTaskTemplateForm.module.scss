.description {
  position: relative;
}

.descriptionLength {
  font-size: 0.75rem;
  letter-spacing: 0.02rem;
  line-height: 1rem;
  position: absolute;
  top: 0;
  right: 0;
}
