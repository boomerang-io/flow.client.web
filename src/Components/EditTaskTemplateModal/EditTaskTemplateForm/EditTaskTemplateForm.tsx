//@ts-nocheck
import React from "react";
import PropTypes from "prop-types";
import { Formik } from "formik";
import * as Yup from "yup";
import orderBy from "lodash/orderBy";
import {
  Button,
  Creatable,
  ModalBody,
  ModalFooter,
  ModalForm,
  TextInput,
  TextArea,
} from "@boomerang-io/carbon-addons-boomerang-react";
import SelectIcon from "Components/SelectIcon";
import { taskIcons } from "Utils/taskIcons";
import styles from "./EditTaskTemplateForm.module.scss";

EditTaskTemplateForm.propTypes = {
  closeModal: PropTypes.func,
  handleEditTaskTemplateModal: PropTypes.func,
  nodeType: PropTypes.string,
  taskTemplates: PropTypes.array,
  templateData: PropTypes.object,
};

function EditTaskTemplateForm({ closeModal, handleEditTaskTemplateModal, nodeType, taskTemplates, templateData }) {
  let taskTemplateNames = taskTemplates
    .map((taskTemplate) => taskTemplate.name)
    .filter((templateName) => templateName !== templateData.name);
  const orderedIcons = orderBy(taskIcons, ["name"]);
  const selectedIcon = orderedIcons.find((icon) => icon.name === templateData.icon);
  const handleSubmit = async (values) => {
    await handleEditTaskTemplateModal({ newValues: values });
    closeModal();
  };

  const formattedEnvs = templateData.envs.map((env) => {
    return `${env.name}:${env.value}`;
  });

  return (
    <Formik
      initialValues={{
        name: templateData.name,
        category: templateData.category,
        icon: selectedIcon ? { value: selectedIcon.name, label: selectedIcon.name, icon: selectedIcon.icon } : {},
        description: templateData.description,
        arguments: templateData.arguments,
        command: templateData.command,
        image: templateData.image,
        nodeType: nodeType,
        script: templateData.script,
        workingDir: templateData.workingDir,
        envs: formattedEnvs,
        serviceAccountName: templateData.serviceAccountName,
        securityContext: templateData.securityContext,
      }}
      validationSchema={Yup.object().shape({
        name: Yup.string()
          .required("Name is required")
          .notOneOf(taskTemplateNames, "Enter a unique value for task name"),
        category: Yup.string().required("Enter a category"),
        icon: Yup.object().shape({
          value: Yup.string().required(),
          label: Yup.string().required(),
        }),
        description: Yup.string()
          .lowercase()
          .min(4, "Description must be at least four characters")
          .max(200, "Description must be less than 200 characters")
          .required("Description is required"),
        arguments: Yup.string().nullable(),
        command: Yup.string().nullable(),
        image: Yup.string().nullable(),
        script: Yup.string().nullable(),
        workingDir: Yup.string().nullable(),
        serviceAccountName: Yup.string().nullable(),
        securityContext: Yup.string().nullable(),
      })}
      onSubmit={handleSubmit}
      initialErrors={[{ name: "Name required" }]}
    >
      {(props) => {
        const { handleSubmit, isValid, values, errors, touched, handleChange, setFieldValue, handleBlur } = props;
        return (
          <ModalForm onSubmit={handleSubmit}>
            <ModalBody aria-label="inputs">
              <TextInput
                id="name"
                invalid={errors.name && touched.name}
                invalidText={errors.name}
                labelText="Name"
                helperText="Must be unique"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="Enter a name"
                value={values.name}
              />
              <TextInput
                id="category"
                invalid={errors.category && touched.category}
                invalidText={errors.category}
                labelText="Category"
                helperText="Categories have strict matching, type as you want to see it"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="e.g. communication"
                value={values.category}
              />
              <SelectIcon
                onChange={({ selectedItem }) => Boolean(selectedItem) && setFieldValue("icon", selectedItem)}
                selectedIcon={values.icon}
                iconOptions={orderedIcons}
              />
              <div className={styles.description}>
                <p className={styles.descriptionLength}>{values.description.length}/200</p>
                <TextArea
                  id="description"
                  invalid={errors.description && touched.description}
                  invalidText={errors.description}
                  labelText="Description"
                  onBlur={handleBlur}
                  onChange={handleChange}
                  value={values.description}
                />
              </div>
              <TextInput
                id="image"
                labelText="Image (optional)"
                helperText="Path to container image"
                name="image"
                value={values.image}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.image && touched.image}
                invalidText={errors.image}
              />
              <TextArea
                id="command"
                labelText="Command (optional)"
                helperText="Override the entry point of the container"
                name="command"
                value={values.command}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.command && touched.command}
                invalidText={errors.command}
              />
              <TextArea
                id="arguments"
                labelText="Arguments (optional)"
                helperText="Enter arguments delimited by a new line"
                placeholder="e.g. system sleep"
                name="arguments"
                value={values.arguments}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.arguments && touched.arguments}
                invalidText={errors.arguments}
              />
              <TextArea
                id="script"
                invalid={errors.script && touched.script}
                invalidText={errors.script}
                labelText="Script (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.script}
              />
              <TextInput
                id="workingDir"
                invalid={errors.workingDir && touched.workingDir}
                invalidText={errors.workingDir}
                labelText="Working Directory (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.workingDir}
              />
              <Creatable
                createKeyValuePair
                id="envs"
                labelText=""
                onChange={(createdItems: string[]) => setFieldValue("envs", createdItems)}
                keyLabelText="Environments (optional)"
                placeholder="Enter env"
                values={values.envs || []}
              />
              <TextInput
                id="serviceAccountName"
                invalid={errors.serviceAccountName && touched.serviceAccountName}
                invalidText={errors.serviceAccountName}
                labelText="Service Account Name (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.serviceAccountName}
              />
              <TextArea
                id="securityContext"
                invalid={errors.securityContext && touched.securityContext}
                invalidText={errors.securityContext}
                labelText="Security Context (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.securityContext}
              />
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" onClick={closeModal} type="button">
                Cancel
              </Button>
              <Button disabled={!isValid} type="submit">
                Update
              </Button>
            </ModalFooter>
          </ModalForm>
        );
      }}
    </Formik>
  );
}

export default EditTaskTemplateForm;
