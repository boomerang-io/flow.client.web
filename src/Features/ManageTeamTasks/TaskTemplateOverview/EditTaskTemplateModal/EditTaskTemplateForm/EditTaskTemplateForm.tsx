//@ts-nocheck
import React from "react";
import PropTypes from "prop-types";
import { Formik } from "formik";
import * as Yup from "yup";
import orderBy from "lodash/orderBy";
import {
  Button,
  ModalBody,
  ModalFooter,
  ModalForm,
  TextInput,
  TextArea,
  Toggle,
} from "@boomerang-io/carbon-addons-boomerang-react";
import SelectIcon from "Components/SelectIcon";
import { taskIcons } from "Utils/taskIcons";
import { NodeType } from "Constants";

EditTaskTemplateForm.propTypes = {
  closeModal: PropTypes.func,
  handleEditTaskTemplateModal: PropTypes.func,
  nodeType: PropTypes.string,
  taskTemplates: PropTypes.array,
  templateData: PropTypes.object,
};

function EditTaskTemplateForm({ closeModal, handleEditTaskTemplateModal, nodeType, taskTemplates, templateData }) {
  let taskTemplateNames = taskTemplates
    .map((taskTemplate) => taskTemplate.name)
    .filter((templateName) => templateName !== templateData.name);
  const orderedIcons = orderBy(taskIcons, ["name"]);
  const selectedIcon = orderedIcons.find((icon) => icon.name === templateData.icon);
  const handleSubmit = async (values) => {
    await handleEditTaskTemplateModal({ newValues: values });
    closeModal();
  };
  return (
    <Formik
      initialValues={{
        name: templateData.name,
        category: templateData.category,
        icon: selectedIcon ? { value: selectedIcon.name, label: selectedIcon.name, icon: selectedIcon.icon } : {},
        description: templateData.description,
        arguments: templateData.arguments,
        command: templateData.command,
        image: templateData.image,
        nodeType: nodeType,
        enableLifecycle: templateData.enableLifecycle,
      }}
      validationSchema={Yup.object().shape({
        name: Yup.string()
          .required("Name is required")
          .notOneOf(taskTemplateNames, "Enter a unique value for task name"),
        category: Yup.string().required("Enter a category"),
        icon: Yup.object().shape({
          value: Yup.string().required(),
          label: Yup.string().required(),
        }),
        description: Yup.string()
          .lowercase()
          .min(4, "Description must be at least four characters")
          .max(200, "Description must be less than 60 characters")
          .required("Description is required"),
        arguments: Yup.string().when("nodeType", {
          is: (nodeType) => nodeType !== NodeType.CustomTask,
          then: Yup.string().required("Arguments are required"),
        }),
        command: Yup.string().nullable(),
        image: Yup.string().nullable(),
        enableLifecycle: Yup.boolean(),
      })}
      onSubmit={handleSubmit}
      initialErrors={[{ name: "Name required" }]}
    >
      {(props) => {
        const { handleSubmit, isValid, values, errors, touched, handleChange, setFieldValue, handleBlur } = props;
        return (
          <ModalForm onSubmit={handleSubmit}>
            <ModalBody aria-label="inputs">
              <TextInput
                id="name"
                invalid={errors.name && touched.name}
                invalidText={errors.name}
                labelText="Name"
                helperText="Must be unique"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="Enter a name"
                value={values.name}
              />
              <TextInput
                id="category"
                invalid={errors.category && touched.category}
                invalidText={errors.category}
                labelText="Category"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="e.g. communication"
                value={values.category}
              />
              <SelectIcon
                onChange={({ selectedItem }) => Boolean(selectedItem) && setFieldValue("icon", selectedItem)}
                selectedIcon={values.icon}
                iconOptions={orderedIcons}
              />
              <TextArea
                id="description"
                invalid={errors.description && touched.description}
                invalidText={errors.description}
                labelText="Description"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.description}
              />
              <TextInput
                id="arguments"
                labelText="Arguments"
                helperText="Enter arguments delimited by a space character"
                placeholder="e.g. system sleep"
                name="arguments"
                value={values.arguments}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.arguments && touched.arguments}
                invalidText={errors.arguments}
              />
              <TextInput
                id="image"
                labelText="Image (optional)"
                helperText="Path to container image"
                name="image"
                value={values.image}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.image && touched.image}
                invalidText={errors.image}
              />
              <TextInput
                id="command"
                labelText="Command (optional)"
                helperText="Override the entry point of the container"
                name="command"
                value={values.command}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.command && touched.command}
                invalidText={errors.command}
              />
              <Toggle
                id="enableLifecycle"
                labelText="Enable Lifecycle"
                helperText="Enable to create lifecycle init and watcher containers to watch for result parameters"
                name="enableLifecycle"
                toggled={values.enableLifecycle}
                onBlur={handleBlur}
                onChange={handleChange}
              />
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" onClick={closeModal} type="button">
                Cancel
              </Button>
              <Button disabled={!isValid} type="submit">
                Update
              </Button>
            </ModalFooter>
          </ModalForm>
        );
      }}
    </Formik>
  );
}

export default EditTaskTemplateForm;
