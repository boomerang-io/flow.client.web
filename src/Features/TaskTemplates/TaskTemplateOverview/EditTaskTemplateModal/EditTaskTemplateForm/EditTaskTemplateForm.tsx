//@ts-nocheck
import React from "react";
import PropTypes from "prop-types";
import { Formik } from "formik";
import * as Yup from "yup";
import orderBy from "lodash/orderBy";
import {
  Button,
  Creatable,
  ModalBody,
  ModalFooter,
  ModalForm,
  TextInput,
  TextArea,
} from "@boomerang-io/carbon-addons-boomerang-react";
import SelectIcon from "Components/SelectIcon";
import { taskIcons } from "Utils/taskIcons";

EditTaskTemplateForm.propTypes = {
  closeModal: PropTypes.func,
  handleEditTaskTemplateModal: PropTypes.func,
  nodeType: PropTypes.string,
  taskTemplates: PropTypes.array,
  templateData: PropTypes.object,
};

function EditTaskTemplateForm({ closeModal, handleEditTaskTemplateModal, nodeType, taskTemplates, templateData }) {
  let taskTemplateNames = taskTemplates
    .map((taskTemplate) => taskTemplate.name)
    .filter((templateName) => templateName !== templateData.name);
  const orderedIcons = orderBy(taskIcons, ["name"]);
  const selectedIcon = orderedIcons.find((icon) => icon.name === templateData.icon);
  const handleSubmit = async (values) => {
    await handleEditTaskTemplateModal({ newValues: values });
    closeModal();
  };

  const formattedEnvs = templateData.envs.map((env) => {
    return `${env.name}:${env.value}`;
  });

  return (
    <Formik
      initialValues={{
        name: templateData.name,
        category: templateData.category,
        icon: selectedIcon ? { value: selectedIcon.name, label: selectedIcon.name, icon: selectedIcon.icon } : {},
        description: templateData.description,
        arguments: templateData.arguments,
        command: templateData.command,
        image: templateData.image,
        nodeType: nodeType,
        script: templateData.script,
        workingDir: templateData.workingDir,
        envs: formattedEnvs,
      }}
      validationSchema={Yup.object().shape({
        name: Yup.string()
          .required("Name is required")
          .notOneOf(taskTemplateNames, "Enter a unique value for task name"),
        category: Yup.string().required("Enter a category"),
        icon: Yup.object().shape({
          value: Yup.string().required(),
          label: Yup.string().required(),
        }),
        description: Yup.string()
          .lowercase()
          .min(4, "Description must be at least four characters")
          .max(200, "Description must be less than 60 characters")
          .required("Description is required"),
        arguments: Yup.string().nullable(),
        command: Yup.string().nullable(),
        image: Yup.string().nullable(),
        script: Yup.string().nullable(),
        workingDir: Yup.string().nullable(),
      })}
      onSubmit={handleSubmit}
      initialErrors={[{ name: "Name required" }]}
    >
      {(props) => {
        const { handleSubmit, isValid, values, errors, touched, handleChange, setFieldValue, handleBlur } = props;
        return (
          <ModalForm onSubmit={handleSubmit}>
            <ModalBody aria-label="inputs">
              <TextInput
                id="name"
                invalid={errors.name && touched.name}
                invalidText={errors.name}
                labelText="Name"
                helperText="Must be unique"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="Enter a name"
                value={values.name}
              />
              <TextInput
                id="category"
                invalid={errors.category && touched.category}
                invalidText={errors.category}
                labelText="Category"
                helperText="Categories have strict matching, type as you want to see it"
                onBlur={handleBlur}
                onChange={handleChange}
                placeholder="e.g. communication"
                value={values.category}
              />
              <SelectIcon
                onChange={({ selectedItem }) => Boolean(selectedItem) && setFieldValue("icon", selectedItem)}
                selectedIcon={values.icon}
                iconOptions={orderedIcons}
              />
              <TextArea
                id="description"
                invalid={errors.description && touched.description}
                invalidText={errors.description}
                labelText="Description"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.description}
              />
              <TextArea
                id="arguments"
                labelText="Arguments (optional)"
                helperText="Enter arguments delimited by a new line"
                placeholder="e.g. system sleep"
                name="arguments"
                value={values.arguments}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.arguments && touched.arguments}
                invalidText={errors.arguments}
              />
              <TextInput
                id="workingDir"
                invalid={errors.workingDir && touched.workingDir}
                invalidText={errors.workingDir}
                labelText="Working Directory (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.workingDir}
              />
              <TextInput
                id="image"
                labelText="Image (optional)"
                helperText="Path to container image"
                name="image"
                value={values.image}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.image && touched.image}
                invalidText={errors.image}
              />
              <TextArea
                id="command"
                labelText="Command (optional)"
                helperText="Override the entry point of the container"
                name="command"
                value={values.command}
                onBlur={handleBlur}
                onChange={handleChange}
                invalid={errors.command && touched.command}
                invalidText={errors.command}
              />
              <TextArea
                id="script"
                invalid={errors.script && touched.script}
                invalidText={errors.script}
                labelText="Script (optional)"
                onBlur={handleBlur}
                onChange={handleChange}
                value={values.script}
              />
              <Creatable
                createKeyValuePair
                id="envs"
                onChange={(createdItems: string[]) => setFieldValue("envs", createdItems)}
                keyLabelText="Environments (optional)"
                placeholder="Enter env"
                values={values.envs || []}
              />
            </ModalBody>
            <ModalFooter>
              <Button kind="secondary" onClick={closeModal} type="button">
                Cancel
              </Button>
              <Button disabled={!isValid} type="submit">
                Update
              </Button>
            </ModalFooter>
          </ModalForm>
        );
      }}
    </Formik>
  );
}

export default EditTaskTemplateForm;
