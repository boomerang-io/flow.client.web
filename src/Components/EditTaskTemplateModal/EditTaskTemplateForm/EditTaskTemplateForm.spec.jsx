import { vi } from "vitest";
import EditTaskTemplateForm from "./index";

const mockfn = vi.fn();
const mockRevision = {
  name: "Test name",
  arguments: ["slack"],
  changelog: {
    userId: "1234",
    reason: "",
    date: "2021-03-19T23:34:09.258+00:00",
    userName: "Test User",
  },
  description: "Test description",
  command: [],
  config: [
    {
      required: false,
      placeholder: "https://hooks.slack.com/services/...",
      defaultValue: "",
      readOnly: false,
      description: "Found within your webhook integration settings",
      key: "url",
      label: "URL",
      type: "text",
      helperText: "",
    },
  ],
  envs: [{ name: "test", value: "TEST" }],
  image: "",
  results: null,
  script: null,
  version: 1,
  workingDir: null,
};
const props = {
  closeModal: mockfn,
  handleEditTaskTemplateModal: mockfn,
  nodeType: "customTask",
  taskTemplates: [mockRevision],
  templateData: mockRevision,
};

describe("EditTaskTemplateForm --- Snapshot", () => {
  it("Capturing Snapshot of Task Templates", async () => {
    const { baseElement } = rtlContextRouterRender(<EditTaskTemplateForm {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
