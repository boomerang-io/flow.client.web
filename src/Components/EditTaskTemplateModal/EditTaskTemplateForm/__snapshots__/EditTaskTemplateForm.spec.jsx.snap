// Vitest Snapshot v1

exports[`EditTaskTemplateForm --- Snapshot > Capturing Snapshot of Task Templates 1`] = `
<body
  id="app"
>
  <div>
    <form
      class="bx--bmrg-modal-form"
    >
      <div
        aria-label="inputs"
        class="bx--modal-content"
      >
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="name"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Name
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  aria-describedby="name-helper-text"
                  class="bx--text-input bx--text-input--md"
                  id="name"
                  placeholder="Enter a name"
                  title="Enter a name"
                  type="text"
                  value="Test name"
                />
              </div>
              <div
                class="bx--form__helper-text"
                id="name-helper-text"
              >
                Must be unique
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="category"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Category
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  aria-describedby="category-helper-text"
                  class="bx--text-input bx--text-input--md"
                  id="category"
                  placeholder="e.g. communication"
                  title="e.g. communication"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="bx--form__helper-text"
                id="category-helper-text"
              >
                Categories have strict matching, type as you want to see it
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-select"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label"
            class="bx--list-box__wrapper"
            role="combobox"
          >
            <label
              class="bx--label"
              for="select-icon"
              id="downshift-0-label"
            >
              <div
                style="display: flex;"
              >
                <div>
                  Icon
                </div>
              </div>
            </label>
            <div
              aria-label="Choose an item"
              class="bx--combo-box bx--list-box"
            >
              <div
                aria-haspopup="true"
                aria-label="open menu"
                class="bx--list-box__field"
                data-toggle="true"
                id="select-icon"
                role="button"
                tabindex="-1"
                type="button"
              >
                <input
                  aria-autocomplete="list"
                  aria-labelledby="downshift-0-label"
                  autocomplete="off"
                  class="bx--text-input bx--text-input--empty"
                  id="select-icon"
                  placeholder="Choose an icon"
                  tabindex="0"
                  type="text"
                  value=""
                />
                
                <div
                  class="bx--list-box__menu-icon"
                >
                  <svg
                    aria-label="Open menu"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    name="chevron--down"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                    />
                    <title>
                      Open menu
                    </title>
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="bx--form__helper-text"
              id="combobox-helper-text-1"
            >
              Choose the icon that best fits this task
            </div>
          </div>
        </div>
        <div>
          <p>
            16
            /200
          </p>
          <div
            class="bx--bmrg-text-area"
          >
            <div
              class="bx--form-item"
            >
              <div
                class="bx--text-area__label-wrapper"
              >
                <label
                  class="bx--label"
                  for="description"
                >
                  <div
                    style="display: flex;"
                  >
                    <div>
                      Description
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="bx--text-area__wrapper"
              >
                <textarea
                  class="bx--text-area"
                  cols="50"
                  id="description"
                  rows="4"
                >
                  Test description
                </textarea>
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="image"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Image (optional)
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  aria-describedby="image-helper-text"
                  class="bx--text-input bx--text-input--md"
                  id="image"
                  name="image"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="bx--form__helper-text"
                id="image-helper-text"
              >
                Path to container image
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-area"
        >
          <div
            class="bx--form-item"
          >
            <div
              class="bx--text-area__label-wrapper"
            >
              <label
                class="bx--label"
                for="command"
              >
                <div
                  style="display: flex;"
                >
                  <div>
                    Command (optional)
                  </div>
                </div>
              </label>
            </div>
            <div
              class="bx--text-area__wrapper"
            >
              <textarea
                class="bx--text-area"
                cols="50"
                id="command"
                name="command"
                rows="4"
              />
            </div>
            <div
              class="bx--form__helper-text"
            >
              Override the entry point of the container
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-area"
        >
          <div
            class="bx--form-item"
          >
            <div
              class="bx--text-area__label-wrapper"
            >
              <label
                class="bx--label"
                for="arguments"
              >
                <div
                  style="display: flex;"
                >
                  <div>
                    Arguments (optional)
                  </div>
                </div>
              </label>
            </div>
            <div
              class="bx--text-area__wrapper"
            >
              <textarea
                class="bx--text-area"
                cols="50"
                id="arguments"
                name="arguments"
                placeholder="e.g. system sleep"
                rows="4"
              >
                slack
              </textarea>
            </div>
            <div
              class="bx--form__helper-text"
            >
              Enter arguments delimited by a new line
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-area"
        >
          <div
            class="bx--form-item"
          >
            <div
              class="bx--text-area__label-wrapper"
            >
              <label
                class="bx--label"
                for="script"
              >
                <div
                  style="display: flex;"
                >
                  <div>
                    Script (optional)
                  </div>
                </div>
              </label>
            </div>
            <div
              class="bx--text-area__wrapper"
            >
              <textarea
                class="bx--text-area"
                cols="50"
                id="script"
                rows="4"
              />
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-text-input"
        >
          <div
            class="bx--form-item bx--text-input-wrapper"
          >
            <label
              class="bx--label"
              for="workingDir"
            >
              <div
                class="bx--bmrg-text-input__label"
              >
                <div>
                  Working Directory (optional)
                </div>
              </div>
            </label>
            <div
              class="bx--text-input__field-outer-wrapper"
            >
              <div
                class="bx--text-input__field-wrapper"
              >
                <input
                  class="bx--text-input bx--text-input--md"
                  id="workingDir"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="bx--bmrg-creatable"
        >
          <div
            class="bx--bmrg-creatable__input"
          >
            <div
              class="bx--bmrg-creatable__key-value-inputs"
            >
              <div
                class="bx--form-item bx--text-input-wrapper"
              >
                <label
                  class="bx--label"
                  for="envs-key"
                >
                  Environments (optional)
                </label>
                <div
                  class="bx--text-input__field-outer-wrapper"
                >
                  <div
                    class="bx--text-input__field-wrapper"
                  >
                    <input
                      class="bx--text-input bx--text-input--md"
                      id="envs-key"
                      style="margin-bottom: 0rem; margin-top: 0px;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <span
                class="bx--bmrg-creatable__colon"
                style="margin-top: 1.75rem;"
              >
                :
              </span>
              <div
                class="bx--form-item bx--text-input-wrapper"
              >
                <div
                  class="bx--text-input__field-outer-wrapper"
                >
                  <div
                    class="bx--text-input__field-wrapper"
                  >
                    <input
                      class="bx--text-input bx--text-input--md"
                      id="envs-value"
                      style="margin-bottom: 0rem; margin-top: 1.5rem;"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <button
              class="bx--bmrg-creatable__button bx--btn bx--btn--md bx--btn--primary bx--btn--disabled"
              disabled=""
              tabindex="0"
              type="button"
            >
              Add
              <svg
                aria-hidden="true"
                aria-label="Add"
                class="bx--btn__icon"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                />
              </svg>
            </button>
          </div>
          <div
            class="bx--bmrg-creatable__tags"
          >
            <div
              aria-label="Clear filter test:TEST"
              class="bx--tag bx--tag--filter bx--tag--teal"
              id="tag-1"
            >
              <span
                class="bx--tag__label"
                title="test:TEST"
              >
                test:TEST
              </span>
              <button
                aria-labelledby="tag-1"
                class="bx--tag__close-icon"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="bx--modal-footer bx--btn-set"
      >
        <button
          class="bx--btn bx--btn--secondary"
          tabindex="0"
          type="button"
        >
          Cancel
        </button>
        <button
          class="bx--btn bx--btn--primary bx--btn--disabled"
          disabled=""
          tabindex="0"
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
  </div>
</body>
`;
